app/main.py

```diff
@@ -1,8 +1,34 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_of_instances = [Person(p["name"], p["age"]) for p in people]
+
+    for person in people:
+        instance = Person.people[person["name"]]
+
+        if check_marital_status(person) == "married":
+            if check_gender(person) == "man":
+                instance.wife = Person.people[person["wife"]]
+            else:
+                instance.husband = Person.people[person["husband"]]
+
+    return list_of_instances
+
+
+def check_gender(person: dict) -> str:
```

Comment:
U can use only one return here. It will be better approach. =)  
if "wife" in person:
        return "man"
    return "woman"